lib/output.js

```diff
@@ -1239,6 +1239,14 @@ function OutputStream(options) {
     DEFPRINT(AST_ObjectKeyVal, function(self, output){
         var key = self.key;
         var quote = self.quote;
+        var print_as_shorthand = self.shorthand &&
+            self.value instanceof AST_Symbol &&
+            self.key == self.value.print_to_string();
```

Comment:
Not when self.value was mangled. If it was mangled I need to call into the Symbol's print function because it gives me the correct string.